File: Java/Duplicate.java
//Given an array of integers containing n + 1 integers where each integer is in the range [1, n] inclusive, find the duplicate number.
import java.util.*;

public class Duplicate {
    public static void main(String[] args) {
        Scanner sc=new Scanner(System.in);
        System.out.println("Enter size of array");
        int n=sc.nextInt();
        int arr[]=new int[n+1];
        System.out.println("Enter elements in the array");
        for(int i=0;i<=n;i++){
            arr[i]=sc.nextInt();
        }
        System.out.println("Duplicate element is :"+findDuplicate(arr, n));

    }
    public static int findDuplicate(int arr[],int n){
        int res=0;
        for(int i=1;i<=n;i++){
             res^=i;
        }
        for(int x:arr){
            res^=x;
        }
        return res;
    }
}

File: C/duplicate.c
#include <stdio.h>

int findDuplicate(int* nums, int numsSize) {
    int slow = nums[0];
    int fast = nums[0];
    
    do {
        slow = nums[slow];
        fast = nums[nums[fast]];
    } while (slow != fast);

    slow = nums[0];
    while (slow != fast) {
        slow = nums[slow];
        fast = nums[fast];
    }

    return slow;
}

int main() {
    int nums[] = {1, 3, 4, 2, 2};
    int size = sizeof(nums) / sizeof(nums[0]);
    
    int duplicate = findDuplicate(nums, size);
    printf("The duplicate number is: %d\n", duplicate);

    return 0;
}
